### ISSUE TITLE
Could you release the FID code for your generated video?

### ISSUE THREAD

**Issue by Iceland-Leo** [Rudrabha/Wav2Lip#22]
The FID is used to evaluate the single image, but for the generated video, how to implement video FID? As we know, video differs from a single image in that it contains timing consistency.

**Comment by prajwalkr** [comment]
Yes, we plan to release the evaluation code within the next couple of weeks. We are cleaning it up at the moment.

**Comment by Iceland-Leo** [comment]
> Yes, we plan to release the evaluation code within the next couple of weeks. We are cleaning it up at the moment.

Have you finished cleaning up the code by now？

**Comment by prajwalkr** [comment]
We have added some of the cleaned up files: https://github.com/Rudrabha/Wav2Lip/tree/master/evaluation/scores. We will add the instructions to use them in 3-4 days. We have multiple deadlines in the next couple of days, thus delaying this release.

**Comment by Iceland-Leo** [comment]
> We have added some of the cleaned up files: https://github.com/Rudrabha/Wav2Lip/tree/master/evaluation/scores. We will add the instructions to use them in 3-4 days. We have multiple deadlines in the next couple of days, thus delaying this release.

OK, but it is just the lip sync metric, what about the FID calculation method in [https://paperswithcode.com/sota/lip-sync-on-lrs2?p=a-lip-sync-expert-is-all-you-need-for-speech](url)?

**Comment by prajwalkr** [comment]
It will be released along with the instructions.

**Comment by Iceland-Leo** [comment]
> It will be released along with the instructions. Reopening the issue until this is complete.

OK, thank you very much!

**Comment by Iceland-Leo** [comment]
> It will be released along with the instructions. Reopening the issue until this is complete.

Have you finished cleaning up the FID code?

**Comment by prajwalkr** [comment]
You can save the generated faces and the GT faces to different folders and use this repo's code: https://github.com/mseitzer/pytorch-fid to calculate the FID. We are currently swamped with a conference deadline, so we are sorry for the delay. Let us know if you face any issues.

**Comment by Iceland-Leo** [comment]
> You can save the generated faces and the GT faces to different folders and use this repo's code: https://github.com/mseitzer/pytorch-fid to calculate the FID. We are currently swamped with a conference deadline, so we are sorry for the delay. Let us know if you face any issues.

You mean that, extract the images of the corresponding two videos(the generate one and the GT) and then calculate fid score using the standard FID algorithm?

**Comment by prajwalkr** [comment]
Save the face crops (generated and actual real faces) in separate folders and then calculate using the standard FID algorithm. Do not save the full frames. Also, do not paste the face back into the video frame and detect faces again. Directly save the generated faces instead.

**Comment by cfernandezlab** [comment]
Hello! Thanks a lot for sharing the code of the paper :)

I tried to reproduce the FID metric for videos as suggested (I'm using the crops stored in _"/syncnet_python/tmp_dir/pytmp/wav2lip/"_, is that correct?), but I'm getting **FID values ~30, instead of ~3-10 as in the paper.**..

Additionally, I noticed that the quality of the crops decreases from the first to the last one even if the command "-qscale:v 2" is used.

Any suggestions?
Thanks in advance!

**Comment by 649459021** [comment]
> Hello! Thanks a lot for sharing the code of the paper :)
> 
> I tried to reproduce the FID metric for videos as suggested (I'm using the crops stored in _"/syncnet_python/tmp_dir/pytmp/wav2lip/"_, is that correct?), but I'm getting **FID values ~30, instead of ~3-10 as in the paper.**..
> 
> Additionally, I noticed that the quality of the crops decreases from the first to the last one even if the command "-qscale:v 2" is used.
> 
> Any suggestions? Thanks in advance!

The audios are randomly sampled. What is the GT ? @prajwalkr 
Is GT the original video input?

**Comment by sangeun99** [comment]
Hi!

I'm also wondering about the ground truth (GT) for running fid code
Did you get any answer for this?
@prajwalkr @Mayur28 @cfernandezlab @Iceland-Leo @649459021 

Thanks in advance

**Comment by thd-ux** [comment]
When comparing with other benchmark models, do we need to ensure that all models are trained on the same dataset? @prajwalkr 
Thanks!
